### PR TITLE
Feat/check viva completion

### PR DESCRIPTION
**Pull request by TeddyAndersson**
## Explain the changes you’ve made
Added support for updating a case with correct data when the application status in viva is set to require completion (Stickprov).

## Explain why these changes are made
These changes handels updating of a submitted case when viva needs an applicant to pass additional completion information.

## Explain your solution
This includes handling events for when a viva application is successfully submitted, checking if the application status is set to require completion and updating the case with information in order to start a completion process.

## How to test the changes?

**Setup environment**
1. Checkout this branch.
2. Deploy the viva-ms service.
3. Add a new test user with the personal number 196001198685 (Vera Toth)
4. Add a new case for this user.
5. Open up CloudWatch for the checkCompletion Lambda in the AWS console

**Setup event to send**
1. Go to the EventBridge in AWS and prepare to send a new event (EventBridge -> eventbuses -> default -> send event) 
2. Add the following input in the send event view:
Event Source: vivaMs.submitApplication
Detail Type: vivaMsSubmitApplicationSuccess
Event detail: {
	"hashedPersonalNumber": "a_hashed_personal_numer",
	"caseKeys": {
		"PK": "pk_for_case",
		"SK": "sk_for_case"
	}
}

**Scenario One**
Send an event on any user that does not have any completion to fullfill (Anyone except Vera Toth).

Result: The case should not update currentFormId or the status

**Scenarion Two**
Send an event on Vera Toth that should have an ongoing completion.

Result: The case should update the currentFormId and the status on the case.

## Anyhting else? (optional)
If you rather have a review in person or have problem with setting up the test environment, just send me a message.

### CODE REVIEW

File: services/viva-ms/lambdas/checkCompletion.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  );
+  const params = {
+    TableName: config.cases.tableName,
+    Key: keys,
+    UpdateExpression: 'set currentFormId = :formId, #status=:completionStatus',
+    ExpressionAttributeNames: {
+      '#status': 'status',
+    },
+    ExpressionAttributeValues: {
+      ':formId': formId,
```
Comment (dannil76) [inline services/viva-ms/lambdas/checkCompletion.js:61]: kanske ska vara konsekvent och skriva currentFormId hela vägen?
completionStatus anväder du hela vägen :)

File: services/viva-ms/lambdas/createVivaCase.js [hunk 2]
```diff
@@ -11,21 +11,37 @@ import { putItem } from '../../../libs/queries';
...
+  const vadaSSMParams = await VADA_SSM_PARAMS;
 
+  const hahsedPersonalNumber = hash.encode(
+    user.personalNumber,
+    vadaSSMParams.hashSalt,
+    vadaSSMParams.hashSaltLength
+  );
   const [applicationStatusError, applicationStatusResponse] = await to(
-    sendApplicationStatusRequest(user.personalNumber)
+    getApplicationStatus(hahsedPersonalNumber, vadaSSMParams)
```
Comment (dannil76) [inline services/viva-ms/lambdas/createVivaCase.js:29]: kanske det går att få bort argument fler än en?
ex. SSPparams skulle ju kunna ligga i helpern?

File: services/viva-ms/lambdas/submitApplication.js [hunk 3]
```diff
@@ -60,21 +82,14 @@ async function sendApplicationsToViva(applicationRequestBody) {
   return vadaCreateRecurrentVivaApplicationResponse.data;
 }
 
-async function getApplicationRequestBody(myCase) {
+async function getApplicationRequestBody(caseItem, personalNumberHashEncoded) {
```
Comment (dannil76) [inline services/viva-ms/lambdas/submitApplication.js:85]: hmm. återigen ett argument för mycket enligt mig.
varför är ssm params inte ok här?

File: services/viva-ms/serverless.yml [hunk 4]
```diff
@@ -109,3 +113,10 @@ functions:
                             { "prefix": "active:submitted" },
                             { "prefix": "active:ongoing" },
                           ]
+  checkCompletion:
+    handler: lambdas/checkCompletion.main
+    events:
+      - eventBridge:
+          pattern:
+            source:
+              - vivams.submitApplication
```
Comment (dannil76) [inline services/viva-ms/serverless.yml:122]: vivams....kanske viva-ms för att spegla servicenamnet?

**Review by dannil76 — CHANGES_REQUESTED** [review]
snakkes!

**Comment by helsingborgsstad** [comment]
Task linked: [CU-d0zj9g Användaren ska få statusuppdatering om hen är uttagen till stickprov](https://app.clickup.com/t/d0zj9g)

File: services/viva-ms/lambdas/checkCompletion.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+  const caseSSMParams = await CASE_SSM_PARAMS;
+  const [updateCaseError, caseItem] = await to(
+    updateCaseCompletionAttributes(caseKeys, caseSSMParams.recurringFormId)
+  );
+  if (updateCaseError) {
+    throw updateCaseError;
+  }
+
+  console.log(caseItem);
```
[the remark this answers is not in the document]
Comment (TeddyAndersson) [reply services/viva-ms/lambdas/checkCompletion.js:45]: Kanske är onödig, men tänkte slänga en log så att man vet att allting har gått fint. Kanske borde vara lite mer utförlig dock.

File: services/viva-ms/serverless.yml (hunk 4 above)
Comment (TeddyAndersson) [reply services/viva-ms/serverless.yml:122]: kanske vivaMs så att vi följer camelCase?

**Review by dannil76 — APPROVED** [review]

**Comment by dannil76** [comment]
Dubbelkolla så att rätt formId blir valt :)

**Review by dannil76 — APPROVED** [review]

**Pull request merged**